face_api_dataset/dataset.py

```diff
@@ -601,9 +612,88 @@ def _open_modality(
             return info["facial_attributes"]["expression"]
 
         if modality == Modality.GAZE:
-            gaze = [
-                info["facial_attributes"]["gaze"]["horizontal_angle"],
-                info["facial_attributes"]["gaze"]["vertical_angle"],
-            ]
-            return np.array(gaze, dtype=np.float64)
+            world_to_cam = np.array(info["camera"]["transform_world2cam"]["mat_4x4"])
+            gaze_left = np.array(info["gaze_values"]["eye_left"]["gaze_vector"] + [0.0])
+            gaze_right = np.array(info["gaze_values"]["eye_right"]["gaze_vector"] + [.0])
+            print(world_to_cam.shape, gaze_left.shape)
+            return np.stack([(world_to_cam @ gaze_left)[:3], (world_to_cam @ gaze_right)[:3]])
+            return np.stack([(gaze_left)[:3], (gaze_right)[:3]])
+
+        if modality == Modality.FACE_BBOX:
+            segment_img, segment_mapping_int = self._read_segments(number, info)
+
+            segment_mapping = info["segments_mapping"]
+            face_seg_idxs = [segment_mapping_int[segment_mapping[s]] for s in self._face_segments]
+            face_mask = np.isin(segment_img, face_seg_idxs).astype(np.uint16)
+
+            def get_bbox(img: np.ndarray):
+                yxs = np.where(img != 0)
+                bbox = np.min(yxs[1]), np.min(yxs[0]), np.max(yxs[1]), np.max(yxs[0])
+                return bbox
+
+            height, width, _ = self._read_rgb(number).shape
+
+            def expand_bbox(bbox: Tuple[int, int, int, int], padding: int):
+                x0, y0, x1, y1 = bbox
+                x0 = max(0, x0 - padding)
+                y0 = max(0, y0 - padding)
+                x1 = min(x1 + padding, width)
+                y1 = min(y1 + padding, height)
+
+                return (x0, y0, x1, y1)
+
+            face_bbox = get_bbox(face_mask)
+            expanded_face_bbox = expand_bbox(face_bbox, self._face_bbox_pad)
+
+            return expanded_face_bbox
+
+        if modality == Modality.CAM_INTRINSICS:
+            return np.array(info["camera"]["intrinsics"])
+
         raise ValueError("Unknown modality")
+
+    def _read_segments(self, number: str, info: Dict[str, Any]) -> Tuple[np.ndarray, np.ndarray]:
+        import cv2
+
+        segments_file = (
+                self._root / f"{number}.{_modality_files(Modality.SEGMENTS)[1]}"
+        )
+        img = cv2.imread(str(segments_file), cv2.IMREAD_UNCHANGED)
+        if img is None:
+            raise ValueError(f"Error reading {segments_file}")
+        if number in self._image_sizes:
+            if self._image_sizes[number] != img.shape[::-1]:
```

Comment:
I think, correct type annotation for `self._image_sizes` is currently in `main` brunch: `Dict[str, Tuple[int, int]]`